### PR TITLE
more ci actions

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
bors try

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,33 +9,76 @@ jobs:
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          target: ${{ matrix.TARGET }}
+
+      - name: Cache Dependencies
+        uses: Swatinem/rust-cache@v1
+        with:
+          key: ${{ matrix.TARGET }}
```
Comment (Emilgardis) [inline .github/workflows/ci.yml:37]: should add a `sharedKey: prebuilt` here, and in the prebuild job the `key` should be `prebuilt`

Comment (burrbull) [reply .github/workflows/ci.yml:37]: No, Prebuilt and Check tasks are independent and must have different caches.

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
bors try

File: .github/workflows/rustfmt.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          override: true
+          components: rustfmt
+
+      - name: Cache Dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
```
Comment (Emilgardis) [inline .github/workflows/rustfmt.yml:28]: why is this a different workflow?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,28 +9,73 @@ jobs:
...
+      matrix:
+        TARGET: [x86_64-unknown-linux-gnu, x86_64-apple-darwin, x86_64-pc-windows-msvc]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
```
Comment (Emilgardis) [inline .github/workflows/ci.yml:30]: this is different to other embedded-wg toolchains, just a note, not an issue

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: .github/workflows/rustfmt.yml (hunk 2 above)
Comment (burrbull) [reply .github/workflows/rustfmt.yml]: done

**Review by Emilgardis — APPROVED** [review]
Nice!

bors r+

**Pull request merged**